collection-scripts/gather_haproxy_config

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+BASE_COLLECTION_PATH="must-gather"
+HAPROXY_CONFIG_PATH=${OUT:-"${BASE_COLLECTION_PATH}/haproxy"}
```

Comment:
Do we want to put this under, ingress-controllers? What if we move away from haproxy in the future?

Reply:
change to `ingress_controllers`